spec/services/reporting/total_user_count_report_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Reporting::TotalUserCountReport do
+  let(:report_date) { Time.zone.today + 1.day }
+  let!(:user) { create(:user) }
+  let(:expected_report) do
+    [
+      ['All-time user count'],
+      [expected_count],
+    ]
+  end
+
+  subject { described_class.new(report_date) }
+
+  describe '#total_user_count_report' do
+    let(:expected_count) { 1 }
+
+    it 'returns the expected data' do
+      expect(subject.total_user_count_report).to eq(expected_report)
+    end
+
+    context 'with a suspended user' do
+      let(:suspended_user) { create(:user, :suspended) }
+      let(:expected_count) { 2 }
+
+      it 'includes the suspended user in the count' do
+        expect(suspended_user).to be_suspended
+        expect(subject.total_user_count_report).to eq(expected_report)
```

Comment:
I'm not super proud of this pattern. I'm essentially testing `total_user_count`, which I made private (because it should be), in a roundabout way. The diff should hopefully make it clear enough if it fails, but it's a little hamfisted.

All that said, I'm not testing these because I think they might break; I'm testing them to make explicit what I intend as the expected behavior. _Should_ a suspended, unconfirmed, or rejected-for-fraud user be included in "Total users all-time", or is that surprising? Given that we have previously received requests for "active users," I am interpreting "Total users all-time" to include non-active users, but will the consumers of our report agree? (I worry that this turns into a large can of worms otherwise.)